redis/_compat.py

```diff
@@ -15,8 +20,12 @@ def select(rlist, wlist, xlist, timeout):
         while True:
             try:
                 return _select(rlist, wlist, xlist, timeout)
-            except InterruptedError:
-                continue
+            except InterruptedError as e:
+                # Python 2 does not define InterruptedError, instead
+                # try to catch an OSError with errno == EINTR == 4.
+                if hasattr(e, 'errno') and (e.errno == 4 or e.errno is None):
```

Comment:
While the comment helps, this could be clearer testing: `e.errno == errno.EINTR`. I believe the `errno` module is in both Py2 and Py3.

Also, in what case does `e.errno is None` mean we should treat the error as an interrupt and try again?